docs/merkletree/treetex/main.go

```diff
@@ -321,14 +325,21 @@ func modifyRangeNodeInfo() error {
 	return nil
 }
 
+var dataFormat = func(id compact.NodeID) string {
+	return fmt.Sprintf("{$leaf_{%d}$}", id.Index)
+}
+
 var nodeFormats = map[string]nodeTextFunc{
 	"address": func(id compact.NodeID) string {
 		return fmt.Sprintf("%d.%d", id.Level, id.Index)
 	},
 	"hash": func(id compact.NodeID) string {
```

Comment:
How come the "hash" function has to handle leaf nodes, but "address" doesn't?

Reply:
Because the "hash" format nodes where level >=1 is of the form H=H(childL|childR), but the Level=0 needs to be H=H(leafN) [and there will be a "-1" level with [leafN]:

![image](https://user-images.githubusercontent.com/7648032/69866932-53019700-129d-11ea-8e46-77d0c2409c66.png)

Whereas the "address" format doesn't have this differentiation:

![image](https://user-images.githubusercontent.com/7648032/69866967-6ca2de80-129d-11ea-910b-8092590922a5.png)

Reply:
SGTM. Put a comment?

Reply:
done!